### ISSUE TITLE
[FEATURE] Override doom-nvim packer dependencies with local clones or forks 

### ISSUE THREAD

**Issue by connorgmeehan** [NTBBloodbath/doom-nvim#251]
## Feature Request

Users should be able to override a plugins source using a table inside of doom_plugins.lua.

This is important for users who want to use a different fork, or a locally developed version of a plugin.

## How would it work
Users should be able to specify a different path that packer will import from.  This could live in `doom_plugins.lua`.
```lua
module.plugin_overrides = {
     { 'nvim-treesitter/nvim-tree-docs', 'mygithubaccount/my-nvim-tree-docs-fork' }, -- Use github link
     { 'nvim-treesitter/nvim-tree-docs', '~/projects/nvim_plugins/my-nvim-tree-docs-fork' } -- Use local path
}
```

## Implementation

Then we wrap the packer use function
```lua
local wrap_packer = function(packer_use)
  return function(spec)
    local source_package = spec[1]
    if utils.has_value(plugin_overrides, source_package) then
       -- ... replace spec[1] with second value from plugin_overrides
       spec.commit = nil -- Disable pinned packer dependency 
    end
    packer_use(spec)
  end
end

local use = wrap_packer(packer.use) 

use({
    'nvim-treesitter/nvim-tree-docs',
    -- ... config etc
})
```
This will swap out the packer dependency path for one specified in `module.plugin_overrides`.

---
This feature request responds to #181

**Comment by molleweide** [comment]
Holy shit cool!!!

**Comment by connorgmeehan** [comment]
Closed due to v4.0.0 alpha changes in `next` branch
